packages/teleterm/src/ui/services/quickInput/quickInputService.ts

```diff
@@ -99,31 +101,25 @@ export class QuickInputService extends Store<State> {
   // Parses the input string and returns AutocompleteResult which includes information about which
   // token we currently show the autocomplete for and what are the autocomplete suggestions (items)
   // to show.
-  //
-  // TODO(ravicious): This function needs to take cursor index into account instead of assuming that
-  // you want to complete only what's at the end of the input string.
-  getAutocompleteResult(input: string): AutocompleteResult {
-    const autocompleteResult =
-      this.quickCommandPicker.getAutocompleteResult(input);
-
-    // Automatically handle some universal edge cases so that each individual picker doesn't have to
-    // care about them.
-    if (autocompleteResult.kind === 'autocomplete.partial-match') {
-      const { targetToken, suggestions } = autocompleteResult;
-      const isEmpty = suggestions.length === 0;
-      // Don't show suggestions if the only suggestion completely matches the target token.
-      const hasSingleCompleteMatch =
-        suggestions.length === 1 && suggestions[0].token === targetToken.value;
-
-      if (isEmpty || hasSingleCompleteMatch) {
-        return {
-          kind: 'autocomplete.no-match',
-          command: autocompleteResult.command,
-        };
-      }
-    }
-
-    return autocompleteResult;
+  parse(input: string): ParseResult {
+    const parseResult = this.quickCommandParser.parse(input);
+
+    // Automatically handle a universal edge case so that each individual suggester doesn't have to
+    // care about it.
+    const getSuggestionsThenHandleEdgeCase = () =>
```

Comment:
I think both the current and proposed solutions introduce sort of the same UX issue. I think it comes from what the user thinks the `QuickInput` is used for. Some people (myself included when I started) may think its _only_ able to enter commands that show from the suggestions menu, when in reality, its just taking whatever you type and putting it into the shell (regardless of suggestions). I don't think thats super clear in the UX but I don't know how to solve it at the moment.

I don't think the current solution is bad if it was made more explicit that the QuickInput is a "type whatever command you want, we'll just help you craft one". 

> Otherwise if I'm on asteroid.earth and I type "tsh ssh ubuntu@ansible", I get no feedback as to whether ansible is a valid name or not.

I think the targeted users would either use the suggestions as they type to craft the command or already know it exists. 

I wonder if this is something we can target in the upcoming usage metrics. Something like, how many users use the QuickInput for _only_ tsh commands we provide, or for any command.